src/Compilers/CSharp/Test/Semantic/Semantics/NameOfTests.cs

```diff
@@ -2373,5 +2366,687 @@ class Attr : System.Attribute { public Attr(string s) {} }";
             CreateCompilation(source, parseOptions: TestOptions.RegularPreview).VerifyDiagnostics(expectedDiagnostics);
             CreateCompilation(source, parseOptions: TestOptions.Regular11).VerifyDiagnostics(expectedDiagnostics);
         }
+
+        [Fact]
+        public void OpenTypeInNameof_Preview()
+        {
+            CreateCompilation("""
+                using System;
+                using System.Collections.Generic;
+
+                var v = nameof(List<>);
+                Console.WriteLine(v);
+                """).VerifyDiagnostics();
```

Comment:
>VerifyDiagnostics(); [](http://example.com/codeflow?start=21&length=20)

Let's also execute the code and observe behavior at runtime. This is probably applicable to other unit-tests targeting success scenarios